packages/flutter_tools/lib/src/commands/build.dart

```diff
@@ -198,8 +198,14 @@ class BuildCommand extends FlutterCommand {
       archive.addFile(_createSnapshotFile(snapshotPath));
     }
 
-    for (_Asset asset in assets)
-      archive.addFile(_createFile(asset.key, asset.base));
+    for (_Asset asset in assets) {
+      ArchiveFile file = _createFile(asset.key, asset.base);
+      if (file == null) {
+        stderr.writeln('Cannot find asset "${asset.key}" in directory "${path.absolute(asset.base)}".');
```

Comment:
I think we were using logging too much early in the life of this tool. Generally, I think all errors should be written to stderr regardless of the logging level. Status messages to the user should probably go to stdout, again, regardless of the logging level. And only diagnostic stuff should go to logging (`about to run process 'foo arg1 arg2'`). This would be for users to use when things go off the rails and they're trying to determine where in the process things are breaking. And we could probably just dial the use of logging back down to use `info`, so there's less guesswork about which things go where.